src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/delete_sub_menu_request.cc

```diff
@@ -193,48 +213,115 @@ void DeleteSubMenuRequest::on_event(const event_engine::Event& event) {
   const smart_objects::SmartObject& message = event.smart_object();
 
   switch (event.id()) {
-    case hmi_apis::FunctionID::UI_DeleteSubMenu: {
-      EndAwaitForInterface(HmiInterfaces::HMI_INTERFACE_UI);
-      hmi_apis::Common_Result::eType result_code =
-          static_cast<hmi_apis::Common_Result::eType>(
-              message[strings::params][hmi_response::code].asInt());
-      std::string response_info;
-      GetInfo(message, response_info);
-      const bool result = PrepareResultForMobileResponse(
-          result_code, HmiInterfaces::HMI_INTERFACE_UI);
-
-      ApplicationSharedPtr application =
-          application_manager_.application(connection_key());
-
-      if (!application) {
-        SDL_LOG_ERROR("NULL pointer");
-        return;
+    case hmi_apis::FunctionID::UI_DeleteCommand: {
+      SDL_LOG_DEBUG("Received UI_DeleteCommand response");
+
+      const auto corr_id =
+          message[strings::params][strings::correlation_id].asUInt();
+      if (pending_request_corr_id_ == corr_id) {
+        auto msg_params = requests_list_.front();
+
+        const auto result_code = static_cast<hmi_apis::Common_Result::eType>(
+            message[strings::params][hmi_response::code].asInt());
+
+        auto app = application_manager_.application(
+            msg_params[strings::app_id].asUInt());
```

Comment:
```suggestion
        auto app = application_manager_.application(connection_key());
```
While they should act the same, I think it's safer and easier to use the original connection key here

Reply:
@jacobkeeler good point. Fixed in https://github.com/smartdevicelink/sdl_core/pull/3748/commits/b4ba2645b0f6cad95d3ba4ad7c20464fd0bf41b2